README.md

```diff
@@ -1,55 +1,301 @@
 
-Enter the `componenttest` directory and run `pytest`. This should run everything.
+#### GetAutofill
 
-## Helpful Resources
+Request Format:
 
-- [GoogleTest User's Guide](https://google.github.io/googletest/)
-- [Pytest Documentation](https://docs.pytest.org/en/stable/index.html)
-- [Makefile Tutorial](https://makefiletutorial.com/)
+``` 
+{
+  "num_suggestions": <Number of suggestions requested>,
+  "partial_query": <The query which will be autofilled>
+}
+```
+
+- **num_suggestions**: The maximum number of suggestions you would like in response. We may respond with any number of autofill suggestions less than or equal to this.
+- **partial_query**: The partial query you would like autofill responses to. This may also be an empty string, if just the top suggestions are wanted.
+
+Response Format:
+```
+{
+  "suggestions": [ <A list of suggestions for the complete query> ]
+}
+```
+> We do not need to include the number of suggestions we are sending, that is inferred by the size of the list
+
+> Should we wrap responses with the same JSON we expect as well? That probably makes the most sense
+
+Side Effects:
+
+None
+
+#### GetQueryID
+
+Request Format:
+```
+{}
+```
+All necessary information is in the wrapper JSON
+
+Response Format:
+```
+{
+  "query_ID": <Unique identifier of the query>
+}
+```
+- **query_ID**: A number, which is guaranteed to be unique between all calls of this function, even if the Evaluation component crashes.
+
+> Number? Hash? String? Defaulting to just a monotonically increasing number. 0, 1, 2, 3, etc...
+
+Side Effects:
+
+This query ID will never again be returned by `GetQueryID`.
+
+#### ReportSearchResults
+
+Request Format:
+```
+{
+  "query_ID": <Unique identifier of the query>
+  "raw_query": <The exact query the user entered>,
+  "results": [ <A list of results shown to the user> ],
+  "clicked": <Index of the result the user chose in the above list>,
+  "query_timestamp": <A timestamp for the query>
+}
+```
+- **query_ID**: Unique identifier of the query, **which must be generated by a previous call of `GetQueryID`**. This will not be checked, however the query will not be stored if there is a conflict.
+- **raw_query**: This should be what the user types in the search bar, with the exception of a "did you mean", in which case the corrected query should be used. This is ultimately up to the UI team's discretion, as it will be used as the dataset that informs autofill.
+- **results**: A list of the results shown to the user, in order of their display. This can be used to infer which links were clicked, and which were ignored.
+> How does webgraph deal with this? Maybe their list is the links the user "previews" by clicking the node without actually following the link. This gets weird if they are re-querying and doing new subgraphs somehow
```

Comment:
Webgraph UI can calculate this order based on "previewing" nodes, and if they don't, not a huge deal